### ISSUE TITLE
directory gets created outside of messagesDir

### ISSUE THREAD

**Issue by bartvde** [yahoo/babel-plugin-react-intl#23]
I have an app that links to an SDK using npm link. This all works fine, except for babel-plugin-react-intl generating a directory outside of messagesDir.

So the app.json gets created in the correct location:

```build/messages/app.json```

but the .json files of the SDK (which is in node_modules/boundlessgeo-sdk) get generated in sdk in the root of the repo:

```
Barts-MacBook-Air:tabbed bartvandeneijnden$ ls -l sdk/js/components/
total 112
-rw-r--r--  1 bartvandeneijnden  staff  1265 Dec  4 09:06 AddLayer.json
-rw-r--r--  1 bartvandeneijnden  staff   157 Dec  4 09:06 Chart.json
-rw-r--r--  1 bartvandeneijnden  staff  1417 Dec  4 09:06 FeatureTable.json
-rw-r--r--  1 bartvandeneijnden  staff   178 Dec  4 09:06 Geocoding.json
-rw-r--r--  1 bartvandeneijnden  staff   154 Dec  4 09:06 GeocodingResults.json
-rw-r--r--  1 bartvandeneijnden  staff   314 Dec  4 09:06 Geolocation.json
-rw-r--r--  1 bartvandeneijnden  staff   124 Dec  4 09:06 HomeButton.json
-rw-r--r--  1 bartvandeneijnden  staff   275 Dec  4 09:06 ImageExport.json
-rw-r--r--  1 bartvandeneijnden  staff  2010 Dec  4 09:06 LayerListItem.json
-rw-r--r--  1 bartvandeneijnden  staff   568 Dec  4 09:06 Measure.json
-rw-r--r--  1 bartvandeneijnden  staff   152 Dec  4 09:06 QGISLegend.json
-rw-r--r--  1 bartvandeneijnden  staff  1044 Dec  4 09:06 QGISPrint.json
-rw-r--r--  1 bartvandeneijnden  staff  1347 Dec  4 09:06 QueryBuilder.json
-rw-r--r--  1 bartvandeneijnden  staff   285 Dec  4 09:06 Select.son
```

Any pointers to as why this might be happening? This is only when using npm link between app and SDK.

**Comment by bartvde** [comment]
ah it seems it's actually following the symlinks:

```
/Users/bartvandeneijnden/opengeo/git/sdk/js/components/Geocoding.jsx
../../sdk/js/components/Geocoding.jsx
```

**Comment by bartvde** [comment]
Is there a way to protect the path so it never gets out of the messagesDir?

**Comment by bartvde** [comment]
filed as: https://github.com/yahoo/babel-plugin-react-intl/issues/23

**Comment by bartvde** [comment]
I'm just going to add it to .gitignore for now

**Comment by ericf** [comment]
I don't understand the problem. What are the values for: the dir you're running Babel in, the configured `messagesDir`, the dir where the `.js` file is outputted, and the dir where the `.json` files are outputted?

**Comment by bartvde** [comment]
Run babel from: ```/Users/bartvandeneijnden/opengeo/git/sdk-apps/tabbed```
messagesDir: ```./build/messages/```
dir where .js is outputted: ```build```
dir where json files are outputted: ```sdk/js```

**Comment by ericf** [comment]
Sorry I also meant to ask the source path to a JS file. Is it not under `sdk-apps/tabbed/`?

What I'm thinking is that the problem is happening here: https://github.com/yahoo/babel-plugin-react-intl/blob/master/src/index.js#L157-L161

If you can step through those lines while running Babel that might show where the issue is. I haven't seen this happen before so I'm trying to figure out what's different with the inputs to this code that are resulting in the unexpected output.

**Comment by bartvde** [comment]
Some of the source files are under the root dir of the app (sdk-apps/tabbed/), others are in node_modules/boundless-sdk/js/components/ (those are the ones causing the issue. I've stepped through that code block before but will do it again.

**Comment by ericf** [comment]
> I've stepped through that code block before but will do it again.

I've never tried to compile code in `node_modules` with Babel and this plugin, so it's likely something with that which is different. Are you using Webpack or something to run Babel on dependencies?

**Comment by bartvde** [comment]
Here is the output from my logging:

```
 filename: /Users/bartvandeneijnden/opengeo/git/sdk-apps/tabbed/app.jsx
p.relative(process.cwd(), filename): app.jsx
p.dirname(p.relative(process.cwd(), filename)): .
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/Geocoding.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/Geocoding.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/FeatureTable.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/FeatureTable.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/GeocodingResults.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/GeocodingResults.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/QueryBuilder.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/QueryBuilder.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/Select.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/Select.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/Measure.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/Measure.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/Chart.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/Chart.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/Geolocation.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/Geolocation.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/ImageExport.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/ImageExport.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/QGISLegend.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/QGISLegend.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/QGISPrint.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/QGISPrint.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/AddLayer.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/AddLayer.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/HomeButton.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/HomeButton.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
filename: /Users/bartvandeneijnden/opengeo/git/sdk/js/components/LayerListItem.jsx
p.relative(process.cwd(), filename): ../../sdk/js/components/LayerListItem.jsx
p.dirname(p.relative(process.cwd(), filename)): ../../sdk/js/components
```

**Comment by bartvde** [comment]
I'm using babelify with this transform:

```
  "browserify": {
    "transform": [
      "cssify",
      "babelify"
    ]
  }
```

**Comment by bartvde** [comment]
This is a directory listing of the app's node modules:

```
Barts-MacBook-Air:tabbed bartvandeneijnden$ ls -l node_modules/
total 24
drwxr-xr-x  10 bartvandeneijnden  staff  340 Dec  4 09:33 babel-plugin-react-intl
drwxr-xr-x   9 bartvandeneijnden  staff  306 Dec  3 14:56 babelify
lrwxr-xr-x   1 bartvandeneijnden  staff   71 Dec  3 15:05 boundless-sdk -> ../../../../../.nvm/versions/node/v4.2.2/lib/node_modules/boundless-sdk
drwxr-xr-x  13 bartvandeneijnden  staff  442 Dec  3 14:57 browserify
drwxr-xr-x   9 bartvandeneijnden  staff  306 Dec  3 14:56 cssify
drwxr-xr-x  10 bartvandeneijnden  staff  340 Dec  3 14:56 envify
drwxr-xr-x  10 bartvandeneijnden  staff  340 Dec  3 14:56 minifyify
lrwxr-xr-x   1 bartvandeneijnden  staff   68 Dec  3 16:34 openlayers -> ../../../../../.nvm/versions/node/v4.2.2/lib/node_modules/openlayers
drwxr-xr-x   7 bartvandeneijnden  staff  238 Dec  3 14:56 pui-react-buttons
drwxr-xr-x   8 bartvandeneijnden  staff  272 Dec  3 14:56 pui-react-tabs
lrwxr-xr-x   1 bartvandeneijnden  staff   63 Dec  3 15:05 react -> ../../../../../.nvm/versions/node/v4.2.2/lib/node_modules/react
drwxr-xr-x   9 bartvandeneijnden  staff  306 Dec  3 14:56 react-dom
drwxr-xr-x  10 bartvandeneijnden  staff  340 Dec  3 14:57 react-intl
drwxr-xr-x   6 bartvandeneijnden  staff  204 Dec  4 09:05 sdk-debug-server
drwxr-xr-x   9 bartvandeneijnden  staff  306 Dec  3 14:56 through
drwxr-xr-x  12 bartvandeneijnden  staff  408 Dec  3 14:56 watchify
```

The symbolic links are because of the usage of ```npm link```

When I don't use a link, all is fine. However, I need to use a link so I can edit the SDK node module from the app during development.

**Comment by ericf** [comment]
If you can edit the plugin code locally to get it to work within your setup, feel free to send a PR. When I get a chance I'll try to setup a simple project to reproduce that uses symlinks to see if I can reproduce.

**Comment by ericf** [comment]
This should now be fixed with the changes from #32

**Comment by bartvde** [comment]
thanks!
